### PR TITLE
Remove python 2.7 & 3.5 from test matrix. remove pycrypto backend

### PR DESCRIPTION
**Pull request by asherf**
If this is accepted, I can follow up with a PR to remove the python 2 specific stuff (usage of six, future imports, etc...)

I am not that familiar with the internal details of the implementation, so please make sure I made the correct changes especially around the tests suite (skipping tests and deleting some tests).
Happy to update this PR w/ any feedback.

### CODE REVIEW

File: jose/backends/__init__.py [hunk 1]
```diff
@@ -10,19 +10,9 @@
...
-        # in 3.8+
-        # noinspection PyUnresolvedReferences
-        import time
-        if not hasattr(time, "clock"):
-            time.clock = time.perf_counter
-
-    except ImportError:
         from jose.backends.rsa_backend import RSAKey  # noqa: F401
+    except ImportError:
+        RSAKey = None
```
Comment (asherf) [inline jose/backends/__init__.py:15]: This is new behavior. I assumed this library needs to support this (w/ reduced functionality, obviously)

**Comment by asherf** [comment]
The code coverage issue is because there are [conditional imports](https://github.com/mpdavis/python-jose/blob/82bd8aa1987fb574830c73f2631fbf7be22fa3d9/jose/jwe.py#L11) (for python 2/3) that we don't fully cover now (since we don't run python 2) for example:

Screen Shot 2021-03-09 at 7 55 14 AM

As I mentioned, I have a branch where those get removed and I can follow up on that as soon as this PR gets accepted/merged.

**Comment by asherf** [comment]
@blag @dreid I would appreciate any feedback you have on this! thanks!

**Comment by asherf** [comment]
@mpdavis @blag @dreid friendly ping.
I would appreciate it if you could review and perhaps merge this. thanks!

**Comment by asherf** [comment]
ping...

**Comment by blag** [comment]
I haven't had time for this, and I don't know when I will have time to come back to this.

It would be nice if this project had more maintainers. @asherf If you have the time to maintain this project, you might want to email @mpdavis and ask to become a maintainer for this project on GitHub, PyPI, and RTD.

**Pull request merged**